### ISSUE TITLE
Inconsistent frametimes on Steam Deck (linux?)

### ISSUE THREAD

**Issue by claytankozmo** [moonlight-stream/moonlight-qt#795]
**READ ME FIRST!**
If you're here because something basic is not working (like gamepad input, video, or similar), it's probably something specific to your setup, so make sure you've gone through the Troubleshooting Guide first: https://github.com/moonlight-stream/moonlight-docs/wiki/Troubleshooting

If you still have trouble with basic functionality after following the guide, join our Discord server where there are many other volunteers who can help (or direct you back here if it looks like a Moonlight bug after all). https://moonlight-stream.org/discord

**Describe the bug**
I have my steam deck connected to my PC, PC is on ethernet, steam deck on 5ghz wifi, and the PC's frametime graph is flat as can be, the network statistics for moonlight on the deck are as good as they get, but the frametime graph (and feel) on the deck is up and down repeatedly. It seems like it goes perfectly smooth for a few seconds then gets erratic.  Disabling V-sync and frame pacing makes it worse.  There are no network jitter percentages or dropped frames reported. 60fps, whether or not I use RTSS to limit to 60, or 65, or higher doesn't seem to make a difference.

Launching via game mode - maybe gamescope compositor is not playing nice with moonlight?

**Steps to reproduce**
No special steps.

**Screenshots**
If applicable, add screenshots to help explain your problem. If the issue is related to video glitching or poor quality, please include screenshots.
![20220520_125528](https://user-images.githubusercontent.com/72308801/169903220-9eca8d20-df66-41dc-94f0-d7840b399159.jpg)
![20220520_130059](https://user-images.githubusercontent.com/72308801/169903238-61f16c65-1671-4d54-a3d6-319198e3221d.jpg)
![20220520_125503](https://user-images.githubusercontent.com/72308801/169903262-668e4374-b905-48b3-accd-9581f7dac0a3.jpg)

**Affected games**
Happens everywhere, even on the desktop when not doing anything.

**Other Moonlight clients**
- Does the issue occur when using Moonlight on iOS or Android?
Not that I can tell - I don't have a proper frametime graph on android.

**Moonlight settings (please complete the following information)**
150MBPS, V-SYNC, Frametime pacing all on.  2560x1600 res (does it on 1280x800 too), and does it even if I lower the bit rate.

**Gamepad-related issues (please complete if problem is gamepad-related)**
- Do you have any gamepads connected to your host PC directly?
- Does the problem still remain if you stream the desktop and use https://html5gamepad.com to test your gamepad?
  - Instructions for streaming the desktop can be found here: https://github.com/moonlight-stream/moonlight-docs/wiki/Setup-Guide

**Client PC details (please complete the following information)**
Steam deck

**Server PC details (please complete the following information)**
 - OS: Windows 10 
 - GeForce Experience version: Newest one, don't know
 - Nvidia GPU driver: 512.15
 - Antivirus and firewall software: Windows Defender and Windows Firewall

**Moonlight Logs (please attach)**
- On Windows, `Moonlight-###.log` files can be found in `%TEMP%`. Simply type that into the File Explorer path field to navigate there.
- On macOS, `Moonlight-###.log` files can be found in `/tmp`. In Finder, press Cmd+Shift+G, then type `/tmp` to navigate there.
- On Linux with the Flatpak, logs print to the terminal when running with the command: `flatpak run com.moonlight_stream.Moonlight`
- On Linux with the Snap, logs print to the terminal when running with the command: `moonlight`
- [message.txt](https://github.com/moonlight-stream/moonlight-qt/files/8758035/message.txt)

**Additional context**
Anything else you think may be relevant to the issue

**Comment by cgutman** [comment]
Frame times aren't expected to be perfectly smooth while streaming with V-sync enabled. The V-blank intervals for client and host aren't in perfect sync and network jitter prevents frames from being delivered exactly at 16.67ms intervals to hit every V-blank period. Making it perfect(ish) would require buffering frames to smooth over periods where a frame arrives a little too late which would increase latency.

V-sync off mode should be pretty close to 16ms though. If it's not, that indicates that perhaps the compositor is not allowing v-sync off behavior correctly.

For example, here are frame metrics on Windows:
![image](https://user-images.githubusercontent.com/2695644/169928281-1327ebc9-4b74-470c-a083-bbb3b493db8a.png)

The present interval (time between frames) is +/- 3 ms, which corresponds closely to network jitter. There will be a little extra variance because encoding, decoding, and network transmission varies a little based on scene complexity (some frames are larger than others), but that's more or less how it should behave.

**Comment by cgutman** [comment]
We discussed this further on Discord and it seems for some reason that VAAPI+EGL has inconsistent frame times under Gamescope on Steam Deck (for this user only?). We were able to identify a workaround of forcing VDPAU with `LIBVA_DRIVER_NAME=none`, but this doesn't work with Gamescope (only on the desktop).

My own AMD Ryzen 3 3300U doesn't have the issue when running the same Flatpak on Ubuntu 20.04, so it may be a bug that is specific to the Deck software or hardware, a driver/kernel change on the Deck, Gamescope, or some particular OS setting adjusted by the user.

I don't think we can go much further here without data from other affected users to try to narrow down the cause.
